tools/codegen/pkg/inventory/generator.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package inventory is used to generate the mixer adapter inventory source
+// file.
+package inventory // import "istio.io/mixer/tools/codegen/pkg/inventory"
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"io"
+	"sort"
+	"text/template"
+
+	"golang.org/x/tools/imports"
+)
+
+var inventoryTmpl = `// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// THIS FILE IS AUTOMATICALLY GENERATED.
+
+package adapter
+
+import(
+		{{range .}}"{{.Path}}"; {{end}}
```

Comment:
since we have the package name, should we use it here. {{.Package}} {{.Path}}.
Just in case two adapters have the same package name (but different full paths), the person who is building the mixer will have the responsibility to provide names to help us disambiguate. example 
prometheus:istio.io.template.prometheus, 
prometheus2:istio.io.template.v2.prometheus